### ISSUE TITLE
NT2023.001 v1.40 - obsItem - problemas na geração

### ISSUE THREAD

**Issue by leandrosantoslss** [ZeusAutomacao/DFe.NET#1459]
boa tarde
estou tendo problemas na geração do xml quando informo a observacao do item, tag dessa ultima NT
Erro ao refletir tipo 'NFe.Classes.NFe'.

 Detalhes:System.InvalidOperationException: Erro ao refletir propriedade 'infNFe'. ---> System.InvalidOperationException: Erro ao refletir tipo 'NFe.Classes.Informacoes.infNFe'. ---> System.InvalidOperationException: Erro ao refletir propriedade 'infAdic'. ---> System.InvalidOperationException: Erro ao refletir tipo 'NFe.Classes.Informacoes.Observacoes.infAdic'. ---> System.InvalidOperationException: Erro ao refletir propriedade 'obsCont'. ---> System.InvalidOperationException: Erro ao refletir tipo 'NFe.Classes.Informacoes.Observacoes.obsCont'. ---> System.InvalidOperationException: Tipos 'NFe.Classes.Informacoes.Observacoes.obsCont' e 'NFe.Classes.Informacoes.Detalhe.Observacao.obsCont' usam nome tipo XML, 'obsCont', no espaço para nomes 'http://www.portalfiscal.inf.br/nfe'. Use atributos XML para especificar um espaço para nome e/ou nome XML para o tipo.
O problema é o seguinte
Hoje está gerando o XML assim:
```
<obsItem>
<obsCont xmlns="Observacao" xCampo="InfoContribuinte">
<xTexto>ICMS monofásico sobre combustíveis cobrado anteriormente conforme Convênio ICMS 199/2022</xTexto>
</obsCont>
</obsItem>
```
com esse xmlns="Observacao"
e o mesmo não pode ir no XML
na clase obsItem está assim:
`[XmlElement(Namespace = nameof(Observacao))]`
Por isso gera o  `xmlns="Observacao"` para não dar conflito com o obsCont e obsFisco do infAdic
se tiro o `[XmlElement(Namespace = nameof(Observacao))]` da o erro reportado acima
moral da história, precisa ter o `[XmlElement(Namespace = nameof(Observacao))]` para gerar o XML com os valores no lugar correto mais não pode gerar com o `xmlns="Observacao"`
Tem que ficar assim
```
<obsItem>
<obsCont xCampo="InfoContribuinte">
<xTexto>ICMS monofásico sobre combustíveis cobrado anteriormente conforme Convênio ICMS 199/2022</xTexto>
</obsCont>
</obsItem>
```

**Comment by marcosgerene** [comment]
![image](https://github.com/ZeusAutomacao/DFe.NET/assets/15945003/286a1cbd-5b92-4741-8ffc-595f6cca7e05)

**Comment by marcosgerene** [comment]
https://github.com/ZeusAutomacao/DFe.NET/commit/4cbb69b6036cd47f546999277356d672b89fefd2

![image](https://github.com/ZeusAutomacao/DFe.NET/assets/15945003/286a1cbd-5b92-4741-8ffc-595f6cca7e05)

**Comment by marcosgerene** [comment]
https://github.com/ZeusAutomacao/DFe.NET/commit/4cbb69b6036cd47f546999277356d672b89fefd2

![image](https://github.com/ZeusAutomacao/DFe.NET/assets/15945003/286a1cbd-5b92-4741-8ffc-595f6cca7e05)

Eu criei os campos, e o @adeniltonbs adicionou esses namespaces, mas isso foi em maio/2022, achei estranho só agora ter problemas. Por favor poste seu XML completo aqui para que eu possa entender.

![image](https://github.com/ZeusAutomacao/DFe.NET/assets/15945003/45fb82ed-a006-4e68-a149-62a07dfc71b5)

**Comment by leandrosantoslss** [comment]
Segue o arquivo XML gerado com o problema mencionado.

[NFe41231129639780000104550010000015001182937402-assinado.zip](https://github.com/ZeusAutomacao/DFe.NET/files/13257479/NFe41231129639780000104550010000015001182937402-assinado.zip)

**Comment by leandrosantoslss** [comment]
@marcosgerene bom dia, chegou a ver alguma coisa referente ao problema que mencionei?
Tentei aqui de várias formas e não consegui, por isso que acabei recorrendo aqui.

**Comment by marcosgerene** [comment]
@leandrosantoslss

Sim, vi seu XML e a correção está diretamente ligada a sua issue. Fiz referência ao @adeniltonbs aqui e chamei ele no chat dos colaboradores. Como existe um commit dele especificamente adicionando isso estou esperando ele se manifestar.

Por favor, abra um PR desfazendo isso, se não conseguir me avise aqui que eu abro e espero um dos outros colaboradores validar.


@robertorp @danilobreda

**Comment by marcosgerene** [comment]
A correção aplicada gerou outra issue, então reverti e vou aguardar o posicionalmento dos outros colaboradores.

**Comment by MauricioSuporte** [comment]
Olá @leandrosantoslss,

Pelo que entendi, o problema está relacionado ao uso do atributo [XmlElement(Namespace = nameof(Observacao))], que está adicionando o namespace xmlns="Observacao" no XML gerado, mas o mesmo não deve ser incluído na tag obsItem.

Sugiro uma solução que envolva a utilização de uma estratégia de customização do XmlSerializer para evitar a inserção do namespace indesejado, sem remover completamente o atributo [XmlElement]. Isso pode ser feito utilizando a classe XmlSerializerNamespaces e definindo explicitamente os namespaces que devem ser utilizados.

Aqui está um exemplo de como você poderia ajustar a serialização:

```
var xmlSerializerNamespaces = new XmlSerializerNamespaces();
xmlSerializerNamespaces.Add(string.Empty, string.Empty); // Remove os namespaces desnecessários

var serializer = new XmlSerializer(typeof(NFe)); // Substitua pelo tipo correspondente
using (var writer = new StreamWriter("output.xml"))
{
    serializer.Serialize(writer, nfeObject, xmlSerializerNamespaces); // nfeObject é o objeto NFe que você está serializando
}
```
Esse código remove os namespaces indesejados do XML gerado, mantendo a estrutura correta das tags sem inserir o xmlns="Observacao".

Por favor, faça um teste com essa abordagem e veja se o XML é gerado corretamente sem o erro de duplicidade de namespace. Caso ainda enfrente problemas, compartilhe o resultado para que possamos ajustar conforme necessário.

**Comment by MauricioSuporte** [comment]
Prezado(a) @leandrosantoslss,

Devido à ausência de novas interações e para manter a organização do nosso repositório, estamos encerrando esta issue. Caso o problema persista ou haja novas informações, sinta-se à vontade para reabri-la ou criar uma nova issue.
